include/dfu/suit_dfu_fetch_source.h

```diff
@@ -16,45 +29,39 @@
- *        as new chunks of data arrive.
- *        If the data needs to be written at some offset (e.g. if the chunks arrive out of order)
- *        @ref suit_dfu_fetch_source_seek can be used to move the write pointer to the given offset.
+ * @details
+ * Fetch source callbacks are registered via suit_dfu_fetch_source_register.
+ * Multiple such callbacks can be registered. The callbacks are called from inside
+ * the SUIT processor until one of them succeeds.
+ * Iteration will be broken if one of registered fetch sources starts to call
+ * @ref suit_dfu_fetch_source_write_fetched_data or @ref suit_dfu_fetch_source_seek.
  *
- *        It is up to the implementer of this callback how the URI is resolved and how
- *        the data is fetched.
- *        If the function fails to fetch the data from the given URI it must return a negative
- *        error code.
+ * The role of the callback is to fetch data from the provided URI and pass it to SUIT
+ * via @ref suit_dfu_fetch_source_write_fetched_data - the data does not have to be passed
+ * all at once, @ref suit_dfu_fetch_source_write_fetched_data can be called multiple times
+ * as new chunks of data arrive.
+ * If the data needs to be written at some offset (e.g. if the chunks arrive out of order)
+ * @ref suit_dfu_fetch_source_seek can be used to move the write pointer to the given offset.
  *
- * @param[in]   uri URI from which the data should be fetched. Fetch source interprets it.
- *                  In case if fetch source is unable to retrieve requested resource, i.e. due to
- *                  unsupported protocol, it shall just fail, avoiding any prior call to
- *                  @ref suit_dfu_fetch_source_write_fetched_data or
- *                  @ref suit_dfu_fetch_source_seek
+ * It is up to the implementer of this callback how the URI is resolved and how
+ * the data is fetched.
+ * If the function fails to fetch the data from the given URI it must return a negative
+ * error code.
  *
- * @param[in]   uri_length Length of the uri, in bytes
+ * @param[in] uri         URI from which the data should be fetched. Fetch source interprets it.
+ *                        In case if fetch source is unable to retrieve requested resource,
+ *                        i.e. due to unsupported protocol, it shall just fail, avoiding any prior
+ *                        call to @ref suit_dfu_fetch_source_write_fetched_data or
+ *                        @ref suit_dfu_fetch_source_seek.
  *
- * @param[in]   session_id Session ID used by the fetch source manager. It shall be passed without
- *                         modification to calls to @ref suit_dfu_fetch_source_write_fetched_data
- *                         and @ref suit_dfu_fetch_source_seek made by this function,
+ * @param[in] uri_length  Length of the uri, in bytes.
  *
+ * @param[in] session_id  Session ID used by the fetch source manager. It shall be passed without
```

Comment:
This parameter requires some description above. Does it mean that the callback implementation must be ready to fetch multiple URIs in parallel? Is there a way to force downloading one by one to simplify the callback implementation?

Reply:
Currently SUIT processor does not support processing requests in parallel, so each instance of it can perform a single fetch operation at a time.
From the fetcher implementer - I cannot assume that a user will not instantiate more than one SUIT processor...

Reply:
I would clearly state in the API docs if the callback must be reentrant. It's not a strong opinion though.